lib/apiservers/portlayer/restapi/handlers/kv_handlers.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"net/http"
+
+	middleware "github.com/go-swagger/go-swagger/httpkit/middleware"
+	"golang.org/x/net/context"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/go-swagger/go-swagger/swag"
+
+	"github.com/vmware/vic/lib/apiservers/portlayer/models"
+	"github.com/vmware/vic/lib/apiservers/portlayer/restapi/operations"
+	"github.com/vmware/vic/lib/apiservers/portlayer/restapi/operations/kv"
+
+	"github.com/vmware/vic/lib/portlayer/store"
+	"github.com/vmware/vic/pkg/kvstore"
+	"github.com/vmware/vic/pkg/trace"
+)
+
+type KvHandlersImpl struct {
+	defaultStore *kvstore.KeyValueStore
+}
+
+func (handler *KvHandlersImpl) Configure(api *operations.PortLayerAPI, handlerCtx *HandlerContext) {
+	api.KvGetValueHandler = kv.GetValueHandlerFunc(handler.GetValueHandler)
+	api.KvPutValueHandler = kv.PutValueHandlerFunc(handler.PutValueHandler)
+	api.KvDeleteValueHandler = kv.DeleteValueHandlerFunc(handler.DeleteValueHandler)
+
+	// Get the APIKV store -- it should always be present since it's
+	// initialized when the portlayer starts
+	s, _ := store.Store(store.APIKV)
+	handler.defaultStore = s
+}
+
+func (handler *KvHandlersImpl) GetValueHandler(params kv.GetValueParams) middleware.Responder {
+	defer trace.End(trace.Begin(params.Key))
+
+	val, err := handler.defaultStore.Get(trace.NewOperation(context.Background(), "GetValue"), params.Key)
```

Comment:
Nice!  Operations!   You may want to add a more meaningful message though than just the name of the func.  Maybe func(arg)?  Then you can trace through the call stack if things time out for a particular operation.  Just an FYI.